addons/mail/models/mail_canned_response.py

```diff
@@ -0,0 +1,84 @@
+
+    _name = "mail.canned.response"
+    _description = "Canned Response"
+    _order = "id desc"
+    _rec_name = "source"
+
+    source = fields.Char(
+        "Shortcut", required=True, index="trigram",
+        help="Canned response that will automatically be substituted with longer content in your messages."
+        " Type ':' followed by the name of your shortcut (e.g. :hello) to use in your messages.",
+    )
+    substitution = fields.Text(
+        "Substitution",
+        required=True,
+        help="Content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.",
+    )
+    description = fields.Char("Description")
+    last_used = fields.Datetime("Last Used", help="Last time this canned_response was used")
+    group_ids = fields.Many2many("res.groups", string="Authorized Groups")
+    is_shared = fields.Boolean(
+        string="Determines if the canned_response is currently shared with other users",
+        compute="_compute_is_shared",
+        store=True,
+    )
+    is_editable = fields.Boolean(
+        string="Determines if the canned response can be edited by the current user",
+        compute="_compute_is_editable"
+    )
+
+    @api.depends("group_ids")
+    def _compute_is_shared(self):
+        for canned_response in self:
+            canned_response.is_shared = bool(canned_response.group_ids)
+
+    @api.depends_context('uid')
+    @api.depends("create_uid")
+    def _compute_is_editable(self):
+        editable = self.env["mail.canned.response"]
+        creating = self.filtered(lambda c: not c.id)
+        updating = self - creating
+        if creating and creating.check_access_rights("create", raise_exception=False):
+            editable += creating._filter_access_rules_python("create")
+        if updating and updating.check_access_rights("write", raise_exception=False):
+            editable += updating._filter_access_rules_python("write")
+        editable.is_editable = True
+        (self - editable).is_editable = False
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        notifications = []
+        for canned_response in res:
+            payload = {"CannedResponse": canned_response._read_format(["id", "source", "substitution"])}
+            notifications.append((self.env.user.partner_id, "mail.record/insert", payload))
+            for group in canned_response.group_ids:
+                notifications.append((group, "mail.record/insert", payload))
+        self.env["bus.bus"]._sendmany(notifications)
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        payload = {"CannedResponse": self._read_format(["id", "source", "substitution"])}
+        notifications = [(self.env.user.partner_id, "mail.record/insert", payload)]
+        for group in self.group_ids:
+            notifications.append((group, "mail.record/insert", payload))
```

Comment:
In batch this does not send to the correct group according to the current canned response